Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/util/CoreHelpUtils.java

```diff
@@ -237,4 +266,86 @@ public static String getPatientIdentifierIso(PatientType patient) {
         return null != map && !map.isEmpty();
     }
 
+    public static String formatDate(String dateFormat, Date date) {
+        if (StringUtils.isNotBlank(dateFormat) && null != date) {
+            return new SimpleDateFormat(dateFormat).format(date);
+        }
+        return "";
+    }
+
+    public static <T> List<T> getList(Enumeration<T> list) {
+        List<T> ret = new ArrayList<>();
+        while (list.hasMoreElements()) {
+            T item = list.nextElement();
+            ret.add(item);
+        }
+        return ret;
+    }
+
+    public static KeyPair generateKeyPair(int keysize, SecureRandom sr) throws NoSuchAlgorithmException {
+        KeyPairGenerator keyGen = KeyPairGenerator.getInstance("RSA");
+        if (null != sr) {
+            keyGen.initialize(keysize, sr);
+        } else {
+            keyGen.initialize(keysize);
+        }
+        return keyGen.generateKeyPair();
+    }
+
+    public static Certificate[] getCertificateChain(Certificate... certs) {
+        return certs;
+    }
+
+    /**
+     * Create a self-signed X.509 Certificate
+     *
+     * @param dn the X.509 Distinguished Name, eg "CN=Test, L=London, C=GB"
+     * @param pair the KeyPair
+     * @param days how many days from now the Certificate is valid for
+     * @param algorithm the signing algorithm, eg "SHA1withRSA"
+     */
+    public static X509Certificate generateCertificate(String dn, KeyPair pair, int days, String algorithm)
+        throws GeneralSecurityException, IOException {
+        PrivateKey privkey = pair.getPrivate();
+        X509CertInfo info = new X509CertInfo();
+        Date fromDate = new Date();
+        Date toDate = new Date(fromDate.getTime() + days * 86400000l);
```

Comment:
While not wrong, I'd like to try to avoid using Date with magic numbers like this. Can use Joda Time or even Calendar would be better than having "days * 86400000l"